17-lifecycle-lab/src/02-cps-with-indicator.jsx

```diff
@@ -1,26 +1,71 @@
-/**
- * Lab 02: CPS monitor with indicator
- *
- * Add an indicator panel to the CPS monitor
- * that shows a green rectangle if user clicks fast enough
- * or a red rectangle if user clicks too slow
- *
- * Decide how to share information between the components
- * and what values to pass when creating the Indicator
- */
-
-import React from 'react';
-
-const Indicator = React.createClass({
-});
-
-export default createClass({
-  render() {
-    return (<div>
-      <button>Click Fast</button>
-      <p>CPS rate: {0}</p>
-      <Indicator />
-    </div>);
-  }
-});
-
+/**
+ * Created by Test1 on 25/12/2016.
+ */
+
+/**
+ * Created by Test1 on 25/12/2016.
+ */
+
+import React from 'react';
+
+export default class CpsWithIndicator extends React.Component {
+
+    constructor() {
+        super();
+        this.state = {
+            message: ""
+            , cnt: 0
+        };
+        setInterval(this.checkFrequency, 1000);
+    }
+
+
+    componentWillMount() {
+        this.setState({cnt : 0});
+        this.timer = setInterval(() => {
+            this.setState({ticks: this.state.ticks + 1});
+        }, 1000);
+    }
+
+    componentWillUnmount() {
+        clearInterval(this.timer);
+    }
+
+    handleClick = (event) => {
+        this.setState({cnt : this.state.cnt += 1});
+    }
+
+    checkFrequency = () => {
+        if (this.state.cnt > 2) {
+            this.setState({message: "too fast!"});
+        } else if (this.state.cnt < 2) {
+            this.setState({message: "too slow...."});
+        }
+        this.setState({cnt : 0});
+    }
+
+
+    render() {
+        return (<div >
+            <input type="button" onClick={this.handleClick} value="click me"/>
+            <label>{this.state.message}</label>
+            <br/>
+            <ColorIndicator cnt={this.state.cnt}/>
+        </div>);
+    }
+
+}
+
+
+function ColorIndicator(props) {
+    //debugger;
+    if (props.cnt > 2) {

```

Comment:
instead of passing counter, pass predicate (boolean) based on which you will render indicator.